base/metrics/persistent_histogram_allocator.cc

```diff
@@ -686,7 +686,7 @@ void GlobalHistogramAllocator::CreateWithLocalMemory(
       std::make_unique<LocalPersistentMemoryAllocator>(size, id, name))));
 }
 
-#if !BUILDFLAG(IS_NACL)
+#if !BUILDFLAG(IS_NACL) && !defined(STARBOARD)
```

Comment:
The below section deals with mostly memory mapped files. I think it's fine to disable for now as we haven't been using those in //base previously, but i'd recommend creating a separate follow-up to have a look if we can make them work. 

In theory at least, Starboard should be able to support memory mapped files on some, but not all platforms.

Reply:
The code compiles but the test failes for this memory mapped files. Going to add comments and add this to b/316198056 as a follow up.